spec/lib/jenkins_pipeline_builder/pull_request_generator_spec.rb

```diff
@@ -1,99 +1,78 @@
-          },
-          '{{name}}-10',
-          '{{name}}-11']
-      }
+      github_site: github_site,
+      git_org: git_org,
+      git_repo_name: git_repo_name,
+      application_name: application_name
     }
   end
-  let(:jobs) do
-    {
-      '{{name}}-10' => {
-        name: '{{name}}-10',
-        type: :'job-template',
-        value: {
-          name: '{{name}}-10',
-          description: '{{description}}',
-          publishers: [{ downstream: { project: '{{job@{{name}}-11}}' } }]
-        }
-      },
-      '{{name}}-11' => {
-        name: '{{name}}-11',
-        type: :'job-template',
-        value: {
-          name: '{{name}}-11',
-          description: '{{description}}' }
-      }
-    }
-  end
-  let(:create_jobs) do
-    [
-      {
-        name: 'pull_req_test-PR5',
-        type: :project,
-        value: {
-          name: 'pull_req_test-PR5',
-          jobs: ['{{name}}-10',
-                 '{{name}}-11'],
-          pull_request_number: '5' }
-      },
-      {
-        name: 'pull_req_test-PR6',
-        type: :project,
-        value: {
-          name: 'pull_req_test-PR6',
-          jobs: ['{{name}}-10',
-                 '{{name}}-11'],
-          pull_request_number: '6'
-        }
-      }
-    ]
+  let(:job_collection) do
+    double('job_collection',
+           defaults: { value: { application_name: application_name } },
+           jobs: [{ value: { scm_branch: 'master' } }])
   end
-  let(:generator) { JenkinsPipelineBuilder::Generator.new }
-  before do
-    # Request to get current pull requests from github
-    stub_request(:any, 'https://www.github.com/api/v3/repos/constantcontact/jenkins_pipeline_builder/pulls').to_return(body: '[{"number": 5,"state": "open","title": "Update README again" },{"number": 6,"state": "open",  "title": "Update README again2"}]')
-    stub_request(:any, 'http://username:password@127.0.0.1:8080/api/json').to_return(body: '{"assignedLabels":[{}],"mode":"NORMAL","nodeDescription":"the master Jenkins node","nodeName":"","numExecutors":2,"description":null,"jobs":[{"name":"PurgeTest-PR1","url":"http://localhost:8080/job/PurgeTest-PR1/","color":"notbuilt" },{"name":"PurgeTest-PR3","url":"http://localhost:8080/job/PurgeTest-PR3/","color":"notbuilt" },{"name":"PurgeTest-PR4","url":"http://localhost:8080/job/PurgeTest-PR4/","color":"notbuilt"}],"overallLoad":{},"primaryView":{"name":"All","url":"http://localhost:8080/" },"quietingDown":false,"slaveAgentPort":0,"unlabeledLoad":{},"useCrumbs":false,"useSecurity":true,"views":[{"name":"All","url":"http://localhost:8080/"}]}')
-  end
-  describe '#initialize' do
-    after(:all) do
-      FileUtils.rm_r 'pull_requests.csv'
+  subject { JenkinsPipelineBuilder::PullRequestGenerator.new params }
+
+  context '#delete_closed_prs' do
+    it 'deletes closed prs' do
```

Comment:
We should have a test for when there are no closed prs to be deleted and when github returns an error.